src/index.js

```diff
@@ -104,11 +90,31 @@ class LoadTestClient {
             lastN = lastN === -1 ? limitedLastN : Math.min(limitedLastN, lastN);
         }
 
-        if (lastN === this.room.getLastN()) {
-            return;
+        let selectedSource;
+
+        if (this.selectedParticipant) {
+            const selectedParticipantTrack = this.room.jvbJingleSession?.peerconnection?.getRemoteTracks(this.selectedParticipant)?.find(track => track.getType() === 'video');
+            if (selectedParticipantTrack) {
+                selectedSource = selectedParticipantTrack.getSourceName();
+            }
         }
 
-        this.room.setLastN(lastN);
+        if (this.room) {
+            if (this.receiverConstraints.lastN !== lastN ||
+                 this.receiverConstraints.defaultConstraints.maxHeight !== newMaxFrameHeight ||
+                 this.receiverConstraints.selectedSources[0] !== selectedSource) {
+                    this.receiverConstraints.lastN = lastN;
+                    this.receiverConstraints.defaultConstraints.maxHeight = newMaxFrameHeight;
+                    if (selectedSource) {
+                        this.receiverConstraints.selectedSources[0] = selectedSource
```

Comment:
The source name needs to be added to `onStageSources` instead of `selectedSources` if we are trying to emulate stage view in jitsi-meet